### ISSUE TITLE
```vagrant reload``` returns timing-out error messages

### ISSUE THREAD

**Issue by rrosenb12** [hashicorp/vagrant#12130]
Vagrant version: 2.2.14
Host operating system: macOS Big Sur, Version 11.1
Guest operating system: VirtualBox 6.1

```ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :

Vagrant.configure("2") do |config|
  config.vm.box = "ubuntu/trusty64"
  config.vm.network "private_network", ip: "192.168.33.10"
  config.vm.synced_folder ".", "/var/www/html", :nfs => {:mount_options => ["dmode=777", "fmode=666"]}
  config.vm.provision "shell", inline: <<-SHELL
    apt-get update
    apt-get install -y apache2
  SHELL
end

```
Basically, what's happening is that if I run ```vagrant reload```, it runs continues in this looping 
```
default: Warning: Remote connection disconnect. Retrying...
default: Warning: Connection reset. Retrying...
```
message. The only way I've been able to remedy this is to completely destroy the box and run ```vagrant up``` again. This will do, but is obviously not ideal.

**Comment by Braunson** [comment]
I've also had this issue all of a sudden and after destroying the box and trying to `vagrant up` again I'm getting it fail at "Checking for host entries" and it drops a traceback.

**Comment by chrisroberts** [comment]
Looks like this may be related: #12130

**Comment by Braunson** [comment]
@chrisroberts This is #12130, perhaps you meant another issue number?

**Comment by chrisroberts** [comment]
For the original issue here, the root cause of the guest hanging on a reload is due to an fstab entry which is not ignored on failure during boot and results in the guest waiting for a key press from the user before continuing the boot process. The errant fstab entry is fixed via #12184 and will be included in the next version. Until the next version is released, this can be worked around by using the `allow_fstab_modification = false` configuration option for the vm:

https://www.vagrantup.com/docs/vagrantfile/machine_settings#config-vm-allow_fstab_modification

For the net-ssh generated stacktrace, I'm still working on a reproduction of the behavior and will be updating #12194 with what I find (@Braunson)

Cheers!
